extension.js

```diff
@@ -251,7 +281,10 @@ function parseMetadataField(data) {
 function removeRemasterText(datastring) {
 	if(!REMOVE_REMASTER_TEXT)
 		return datastring
-
+		
```

Comment:
Move this to the very top of `_buildLabel`

This should return an empty string if there's only one player, `BUTTON_PLACEHOLDER` if there's more

Reply:
`this.playerList` is the list of mpris sources

Reply:
Upon further inspection this statement looks a bit useless or a bit of a bodge.
If the player is paused and `REMOVE_TEXT_PAUSED` is true this function shouldn't be called, but that was my mistake

Reply:
Hang on, what’s the idea behind `BUTTON_PLACEHOLDER` again? Seems to be linked to having multiple players but I just implemented a check to try identify the “Playing” software so not sure if the new method is consistent.

Reply:
`BUTTON_PLACEHOLDER` is there because some applications "steal" the focus of the extension with an empty string. The placeholder string should hint the user to switch to another mpris source if they want.

By the way, i have not tested your new changes yet.

Reply:
Tested your changes. I see two problems:
* When another source starts to play, the extension jumps to the new one, this could be particularly annoying if you are browsing reddit or whatever, although some people will probably prefer this behaviour (new option?)
* When two or more sources are playing simultaneously, you can't switch between them

Reply:
- I wasn't aware of this case. It would indeed switch as the code is currently iterating from last to first. We could iterate from first to last as I assume that the browser would be added at the end of the list and this would then be consistent with current behaviour?
- Yes, when two apps are playing at the same time, the tool won't know which one to pick but this is already the case. I am not sure how often people listen to multiple songs at the same time though. What would be the expected behaviour if multiple sources are playing, I assume that you would need to pick one as I doubt you would want to list them all?

Reply:
ok, upon further review, the _buildLabel() function no longer , upon further review, the _buildLabel() function no longer  hides the text when playback is paused because of the conditions are only met if there are multiple players and `labelstring` is empty which wouldn't normally be the case. 

The reason for cycling through the players to find out if one was actively Playing was so the text could switch automatically to the active player when one is paused and the other one is started. With the current code, this check isn't getting executed as it only runs if `(labelstring.length == 0)`. Arguably this is a separate suggestion to the original "hide text when playback is paused" so we can certainly discuss it separately if needed. I think that now that you have the ability to figure out if an app is playing, you could use this to skip the player returning empty strings and return the info for an actively playing app `this.player = new Player(this.playerList[i])` with non-zero info (I hope this makes sense)? This way you can try to switch automatically and only return `BUTTON_PLACEHOLDER` in edge cases where there is only one player and it is returning an empty string?

I have an updated code which I think goes some way towards adressing the above. It iterates from 0 so hopefully reddit won't be stealing the focus if started after the music player. if you are worried it might still be and issue, we could easily include the string length check within the loop to exclude players returning empty `labelstring`. Let me know if this is of interest and I can push it as commit.

Reply:
i don't know if this applies to the present version of the file. Check and test the last commit and if you believe your implementation it's better, push it. We can always nitpick or revert the changes later.